test/model/message_list_test.dart

```diff
@@ -427,6 +427,53 @@ void main() {
     });
   });
 
+  group('renarrowAndFetch', () {
+    test('smoke', () => awaitFakeAsync((async) async {
+      final channel = eg.stream();
+
+      const narrow = CombinedFeedNarrow();
+      await prepare(narrow: narrow, stream: channel);
+      final messages = List.generate(100,
+        (i) => eg.streamMessage(id: 1000 + i, stream: channel));
+      await prepareMessages(foundOldest: false, messages: messages);
+
+      // …do something unrelated…
+      connection.prepare(json: olderResult(
+        anchor: 1000, foundOldest: false,
+        messages: List.generate(100,
+          (i) => eg.streamMessage(id: 900 + i, stream: channel)),
+      ).toJson());
+      await model.fetchOlder();
+      checkNotified(count: 2);
+
+      final generationBefore = model.generation;
+
+      final newNarrow = ChannelNarrow(channel.streamId);
+      final newAnchor = NumericAnchor(messages[3].id);
+
+      final result = eg.getMessagesResult(
+        anchor: newAnchor,
+        foundOldest: false, foundNewest: false,
+        messages: messages.sublist(3, 5));
+      connection.prepare(json: result.toJson(), delay: Duration(seconds: 1));
+      model.renarrowAndFetch(newNarrow, newAnchor);
+      checkNotifiedOnce();
+      check(model)
+        ..generation.equals(generationBefore + 1)
```

Comment:
Ideally the generation would be an implementation detail of the view-model, not something a test inspects.

I think the substance of this check can be gotten by having a `fetchOlder` call like above, but delaying its result to come in after the `renarrowAndFetch` (and before the latter fetch completes). Then we can check that after the time the `fetchOlder` result would have arrived, the list is still empty.